app/src/main/kotlin/com/wire/android/util/FileUtil.kt

```diff
@@ -34,8 +37,23 @@ suspend fun Uri.toByteArray(context: Context): ByteArray {
     }
 }
 
+fun getWritableImageAttachment(context: Context): Uri {
+    val file = File(context.cacheDir, TEMP_IMG_ATTACHMENT_FILENAME)
+    file.setWritable(true)
+    return FileProvider.getUriForFile(context, BuildConfig.APPLICATION_ID + ".provider", file)
+}
+
+fun getWritableVideoAttachment(context: Context): Uri {
+    val file = File(context.cacheDir, TEMP_VIDEO_ATTACHMENT_FILENAME)
+    file.setWritable(true)
+    return FileProvider.getUriForFile(context, BuildConfig.APPLICATION_ID + ".provider", file)
+}
+
```

Comment:
i know there is 3 time rule :P that when something repeats 3 times it is probably  good idea to make it more generic, but maybe we can move it into one function ?

Reply:
good idea, done !